ui/js/controllers/bugfiler.js

```diff
@@ -0,0 +1,224 @@
+            if(thBugzillaProductObject[failurePathRoot]) {
+                $scope.suggestedProducts.push(thBugzillaProductObject[failurePathRoot][0]);
+            }
+
+            // Look up product suggestions via Bugzilla's api
+            var productSearch = $scope.productSearch;
+
+            if(productSearch) {
+                $http.get("https://bugzilla.mozilla.org/rest/prod_comp_search/" + productSearch + "?limit=5").then(function(request) {
+                    var data = request.data;
+                    $scope.suggestedProducts = [];
+                    for(var i = 0; i < data.products.length;i++) {
+                        if(data.products[i].product && data.products[i].component) {
+                            $scope.suggestedProducts.push(data.products[i].product + " :: " + data.products[i].component);
+                        }
+                    }
+                });
+            }
+        };
+
+        /*
+         *  This is called once intermittent.html's ng-repeat is finished to select the first product listed
+         */
+        $scope.focusProduct = function() {
+            $("#suggestedProducts").children(":first").children(":first").prop("checked", true);
+        };
+
+        /*
+         *  Same as clicking outside of the modal, but with a nice button-clicking feel...
+         */
+        $scope.cancelFiler = function() {
+            $uibModalInstance.dismiss('cancel');
+        };
+
+        /*
+         *  Actually send the gathered information to bugzilla.
+         */
+        $scope.submitFiler = function() {
+            var bugzillaRoot = "https://bugzilla-dev.allizom.org/"; // (prod is "https://bugzilla.mozilla.org/");
+            var summarystring = $scope.modalSummary;
+            var productString = "";
+            var componentString = "";
+            var isProductSelected = false;
+
+            $(':input','#modalForm').attr("disabled",true);
+
+            var productRadios = $("#modalForm input[name='productGroup'");
+            if(productRadios && productRadios.length) {
+                for(var i=0, len=productRadios.length; i<len; i++) {
+                    if(productRadios[i].checked) {
+                        productString += productRadios[i].value.split(" :: ")[0];
+                        componentString += productRadios[i].value.split(" :: ")[1];
+                        isProductSelected = true;
+                        break;
+                    }
+                }
+            } else {
+                if(productRadios && productRadios.checked) {
+                    productString += productRadios.value.split(" :: ")[0];
+                    componentString += productRadios.value.split(" :: ")[1];
+                    isProductSelected = true;
+                }
+            }
+
+            if(!isProductSelected) {
+                alert("Please select (or search and select) a product/component pair to continue");
```

Comment:
rather than ``alert`` we usually use the ``thNotify`` service.  But your call if you feel like this is better.